recipe-client-addon/lib/ClientEnvironment.jsm

```diff
@@ -203,6 +203,10 @@ this.ClientEnvironment = {
       return Utils.keyBy(addons, "id");
     });
 
+    XPCOMUtils.defineLazyGetter(environment, "isFirstRun", () => {
```

Comment:
Just following the style of the rest of the file, really. I'm not sure that the observer would be helpful since I think these getters are only used when evaluating a recipe and not something shield needs to worry about changing out from under it.